src/eigen.jl

```diff
@@ -136,7 +136,7 @@ end
     TA = eltype(A)
 
     @inbounds if A.uplo == 'U'
-        if a[3] == 0 # A is diagonal
+        if a[3] == zero(TA) # A is diagonal
```

Comment:
Is this ever needed? You can compare with 0, won't that work fine? For example no need to create a `zero(DualNumber)` if the type is a dual number.

Reply:
I guess it depends on what is cheaper: converting `0` to a comparable type or constructing the zero element. I just tested one versus the other, and for `Float` it doesn't make any difference (in contrast to my minor improvement above), but for `Dual` the comparison `== zero(Dual)` is indeed more expensive than `== 0`. I guess I'll revert the last commit then?